packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -453,66 +433,29 @@ class DevFS {
     DateTime firstBuildTime,
     bool bundleFirstUpload = false,
     bool bundleDirty = false,
-    Set<String> fileFilter,
     @required ResidentCompiler generator,
     String dillOutputPath,
     @required bool trackWidgetCreation,
     bool fullRestart = false,
     String projectRootPath,
     @required String pathToReload,
+    @required List<String> invalidatedFiles,
   }) async {
     assert(trackWidgetCreation != null);
     assert(generator != null);
-    // Mark all entries as possibly deleted.
-    for (DevFSContent content in _entries.values) {
-      content._exists = false;
-    }
 
-    // Scan workspace, packages, and assets
-    printTrace('DevFS: Starting sync from $rootDirectory');
-    logger.printTrace('Scanning project files');
-    await _scanDirectory(rootDirectory,
-                         recursive: true,
-                         fileFilter: fileFilter);
-    if (fs.isFileSync(_packagesFilePath)) {
-      printTrace('Scanning package files');
-      await _scanPackages(fileFilter);
-    }
-    if (bundle != null) {
+    if (bundle != null && fullRestart) {
```

Comment:
What is the reason for doing this for `fullRestart` only? Don't we need to scan assets for changes when doing hot reload as well? I think you mentioned there might be a limitation currently that we don't pick up newly added assets on hot reload, but isn't that just a bug? All pieces seems to be in place to have that functionality available.

Reply:
Filled https://github.com/flutter/flutter/issues/29184

Reply:
We need to update both the assets and the asset manifest. Right now we have the capability to update the assets but the manifest doesn't update correctly. Its possible this is an oversight or a bug.

Currently the asset _entries remains populated, so hot reload of assets still works.

Reply:
I'm going to add this back for now, because I've found its possible to hot reload assets in the hello_world example. Its not possible in flutter_gallery where I originally tried, and the differences there are number of assets and the usage of package assets.